lib/tessel/deploy.js

```diff
@@ -308,109 +389,94 @@ actions.tarBundle = function(opts) {
     opts.slim = false;
   }
 
+  var includeRules = actions.glob.rules(target, '.tesselinclude');
+  var includeFiles = actions.glob.files(globRoot, includeRules);
+
   if (opts.slim) {
     logs.info('Generating slim build.');
     return new Promise(function(resolve, reject) {
-      actions.glob(target + '/**/*.tesselignore', {
-        dot: true,
-        mark: true,
-      }, function(error, ignoreFiles) {
-        if (error) {
-          return reject(error);
-        }
+      var ignoreRules = actions.glob.rules(target, '.tesselignore');
+      var ignoreFiles = actions.glob.files(globRoot, ignoreRules);
+
+      var fileCount = 0;
+      var entry = path.join(relative, opts.resolvedEntryPoint);
+      var bify = actions.browserify(entry, {
+        builtins: false,
+        commondir: false,
+        browserField: false,
+        detectGlobals: false,
+        ignoreMissing: true
+      });
 
-        var rules = ignoreFiles.reduce(function(rules, ignoreFile) {
-          var dirname = path.dirname(ignoreFile);
-          var patterns = fs.readFileSync(ignoreFile, 'utf8').trim().split(/\r?\n/).reduce(function(patterns, pattern) {
-            pattern = pattern.trim();
+      // Inform browserify bundler instance
+      // of all files to exclude. If a file exists in the
+      // inclusion list, then DO NOT exclude it from the
+      // bundling phase.
+      ignoreFiles.forEach(function(file) {
+        if (includeFiles.length &&
+          includeFiles.indexOf(file) >= 0) {
+          return;
+        }
+        bify.exclude(path.join(globRoot, file));
+      });
 
-            // Ignores empty lines and comments
-            if (pattern && !pattern.match(/^#/)) {
-              patterns.push(path.relative(target, path.join(dirname, pattern)));
+      bify
+        .on('file', function() {
+          fileCount++;
```

Comment:
TODO: add comment note about file counting w/r to its use in optimizing the bundle size